Add default code of conduct

tstenner opened the pull request:
This is a code of conduct applicable to all repositories under the labstreaminglayer organization that haven't adopted a code of conduct yet.

It's based on the Contributor Covenant, and even though I have the impression that we currently don't have any problems I think it's best to clarify what we're expecting from ourselves and others.

tstenner committed:
Add default code of conduct

mgrivich commented:
I would replace 

> In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.

with

> In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a courteous experience for everyone.

I replaced "harassment-free" with "courteous" as it is possible to toxic without fulfilling the definition of harassment. I removed the list because lists of protected classes are problematic. Either people can be abused for reasons that are not described on the list, or the list is not correct because it is incomplete.

dmedine commented:
I second mgrivich's suggestions. 

Thanks for bringing this up, tstenner. Unfortunately such guidelines are necessary even for our little science and engineering project these days.

trmullen commented:
mgrivich's suggested modification is good and I am fine with that language.

tstenner commented:
> I replaced "harassment-free" with "courteous" as it is possible to toxic without fulfilling the definition of harassment.

Even better.

> I removed the list because lists of protected classes are problematic.

That's a hot topic and while I agree that it might be incomplete explicitely stating it is more inclusive.
As a minor change, I suggest we emphasize "everyone" to make it clear we mean everyone:

> In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a courteous experience for *everyone*.

Also, after sleeping over it I'd add a bullet point under "Examples of behavior that contributes to creating a positive environment
include:":

> * In doubt, assume good intentions.

tstenner committed:
Update Code of Conduct

agricolab commented:
Thanks @tstenner. Is there a reason the sentence "We pledge to act and interact in ways that contribute to an open, welcoming, diverse, inclusive, and healthy community." was omitted from the "our pledge" paragraph?

I also want to add that the [CC FAQ](https://www.contributor-covenant.org/faq/) comments on why it prefers a list of protected classes: 
> The Contributor Covenant explicitly lists protected classes for many reasons, such as reminding people to give them appropriate consideration, and assuring people in those protected classes that they are welcome.  

We are all white men with very similar backgrounds attempting to discuss diversity. I believe we should be cautious when we believe we improve a widely adopted CoC that has been developed with the voices of many stakeholders in mind. Considering the explanation of the CC for the list of protected classes, i'd therefore suggest to stick with the list. We could add a "but not limited to", along the lines of: 
>   for everyone, regardless of, but not limited to, age, body size, visible or invisible disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation

mgrivich commented:
> We are all white men with very similar backgrounds attempting to discuss diversity.

We should not be discounting opinions because of the race (white in this case) of the person who says them. Virtually everyone has experienced exclusion in one way or another, including me, and should have a seat at the table when discussing how to be inclusive.

>  I believe we should be cautious when we believe we improve a widely adopted CoC

Proper behavior is not determined democratically. And if we held this to a vote of all people, we cannot be sure that things will break the way that you expect, especially now.

agricolab commented:
I agree with you. My statement, specifically that at least as far as i know, every developer and contributor is a white man, was not meant to exclude someone from the table, but to highlight the current lack of diversity. It should be obvious, that those from "protected classes" are currently underrepresented, and the CC template is the closest we have to get their voice included to this discussion. 

How would you suggest we can satisfy the reasons the CC gives for using a list of protected classes, i.e  "reminding people to give them appropriate consideration" and "assuring people in those protected classes that they are welcome"?

chkothe commented:
Sorry, I'm a bit late to the discussion. I must say I side with Robert Guggenberger here, and I was thinking the same thing. If I were to belittle someone based on, e.g., their perceived level of experience, there would be nothing in our Code of Conduct that the person could point to and/or remind me of, in order to call out my unacceptable behavior. I might be compelled to argue that level of experience has nothing to do with discrimination or that it's not a big deal for xyz reasons. However, with an explicit listing, even tough it may only cover 80% of the cases, it would be a clear call, and I'd have to instantly take it back instead of arguing over what my interpretation of the "everyone" is or not is. 

I do like the ", but not limited to, " extension though, to get the best of both worlds.

Also, to get a better feel of the degree of modification that other adopters of this CoC have made, I've reviewed all the repos that I recognized out of that list (https://www.contributor-covenant.org/adopters/), and for every single repository where I could find a copy of the CoC, the listing in question is intact and hadn't been replaced by a blanket clause. 

Repos for which I found the CoC, all of which retain the clause: Mono, atom, node.js, rubyonrails, react, electron, git, golang, google open source, swift, taiga.io, tensorflow, travis-ci, jenkins, vue.js, kubernetes, linux.

mgrivich commented:
> How would you suggest we can satisfy the reasons the CC gives for using a list of protected classes, i.e "reminding people to give them appropriate consideration"

People don't really need a reminder. People generally know what the rules are, and choose to follow them, or not. The reason for rules is to have a policy to point to when disciplining someone.

> and "assuring people in those protected classes that they are welcome"?

We are welcoming in our daily interactions. People will pay a lot more attention to what we do rather than what our policy statements are.

My position remains that "everyone", and leaving it at that, is more inclusive than "everyone, but especially the following groups".

tstenner commented:
> Is there a reason the sentence "We pledge to act and interact in ways that contribute to an open, welcoming, diverse, inclusive, and healthy community." was omitted from the "our pledge" paragraph?

Github suggests an older version (1.4) of the CC that doesn't include the last sentence.

> > We are all white men with very similar backgrounds attempting to discuss diversity.

> We should not be discounting opinions because of the race (white in this case) of the person who says them.

I think @agricolab rather meant that we have rather uniform backgrounds and so we're unlikely to have a better idea compared to the team beind the CoC.

> If I were to belittle someone based on, e.g., their perceived level of experience, there would be nothing in our Code of Conduct that the person could point to and/or remind me of, in order to call out my unacceptable behavior. I might be compelled to argue that level of experience has nothing to do with discrimination or that it's not a big deal for xyz reasons. However, with an explicit listing, even tough it may only cover 80% of the cases, it would be a clear call, and I'd have to instantly take it back instead of arguing over what my interpretation of the "everyone" is or not is.

I like this example because it's something that might happen when one's not thinking about it. It's not meant to be an exhaustive list, but rather some characteristics that happen to be causes for discrimination, but rather the basics to make it clear. After all, "No shoes, no shirt, no service" is a reminder to dress appropriately that with a clear idea what's expected and it doesn't suggest that leaving pants at home is ok.
As Christian said, it makes it easier for the maintainers too when there's a complaint and the cause is in the list. In other cases not covered by the list, the list can be amended.

> People don't really need a reminder. People generally know what the rules are, and choose to follow them, or not.

People are also accustomed to their cultural background and might perceive some unwanted behaviours as normal.

> The reason for rules is to have a policy to point to when disciplining someone.

That's the reason for some rules; I rather view the CoC like some parts of the German constitution that are neither specific nor imply concrete courses of action (e.g. "Human dignity shall be inviolable. To respect and protect it shall be the duty of all state authority.") but rather specify some basic ideas that concrete rules should adhere to.

> We are welcoming in our daily interactions. People will pay a lot more attention to what we do rather than what our policy statements are.

I'd argue that we have some ideas how to behave and are therefore welcoming; so putting it down in writing doesn't change the behaviour.

> My position remains that "everyone", and leaving it at that, is more inclusive than "everyone, but especially the following groups".

On the other hand, as Christian noted, lots of other organizations have it included so removing it also sends a message.
